lib/src/utils.dart

```diff
@@ -82,3 +63,76 @@ void logDependencyInfractions(String infraction, Iterable<String> dependencies)
   final sortedDependencies = dependencies.toList()..sort();
   logger.warning([infraction, bulletItems(sortedDependencies), ''].join('\n'));
 }
+
+/// Lists the packages with infractions
+List<String> getDependenciesWithPins(Map dependencies) {
+  final List<String> infractions = [];
+  for (String packageName in dependencies.keys) {
+    String version;
+    final packageMeta = dependencies[packageName];
+
+    if (packageMeta is String) {
+      version = packageMeta;
+    } else if (packageMeta is Map) {
+      if (packageMeta.containsKey('version')) {
+        version = packageMeta['version'];
+      } else {
+        // This feature only works for versions, not git refs or paths.
+        continue;
+      }
+    } else {
+      continue; // no version string set
+    }
+
+    final DependencyPinEvaluation evaluation = inspectVersionForPins(version);
+
+    if (evaluation.isPin) {
+      infractions.add('$packageName -- ${evaluation.message}');
```

Comment:
I still think it would be nice to add the version range here since we have it so handy: 
```dart
'$packageName: $version -- ${evaluation.message}'
```

If there's multiple, it reduces the need to cross-reference them, especially if they're printed in CI or something:
```
These packages are pinned in pubspec.yaml:
  * analyzer -- Inclusive max bounds restrict minor bumps and patches.
  * barback -- Inclusive max bounds restrict minor bumps and patches.
  * built_value -- This pin blocks minor bumps.
  * quiver -- Inclusive max bounds restrict minor bumps and patches.
```
vs
```
These packages are pinned in pubspec.yaml:
  * analyzer: >=0.30.0 <=0.31.0 -- Inclusive max bounds restrict minor bumps and patches.
  * barback: >=0.15.2 <=0.15.2+14 -- Inclusive max bounds restrict minor bumps and patches.
  * built_value: >=4.2.0 <5.2.0 -- This pin blocks minor bumps.
  * quiver: >=0.21.4 <=0.28.0 -- Inclusive max bounds restrict minor bumps and patches.
```

Not a must have, but I'd personally like to have it 😸